File: sdk/java/src/test/java/com/flagent/client/model/ConstraintTest.java
package com.flagent.client.model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Model tests for Constraint
 */
class ConstraintTest {

    @Test
    void testConstraint() {
        Constraint model = new Constraint()
            .id(1L)
            .segmentID(10L)
            .property("region")
            .operator(Constraint.OperatorEnum.IN)
            .value("EU,US");
        Assertions.assertNotNull(model);
        Assertions.assertEquals(1L, model.getId());
        Assertions.assertEquals(10L, model.getSegmentID());
        Assertions.assertEquals("region", model.getProperty());
        Assertions.assertEquals(Constraint.OperatorEnum.IN, model.getOperator());
        Assertions.assertEquals("EU,US", model.getValue());
    }

    @Test
    void idTest() {
        Constraint model = new Constraint().id(5L);
        Assertions.assertEquals(5L, model.getId());
    }

    @Test
    void segmentIDTest() {
        Constraint model = new Constraint().segmentID(20L);
        Assertions.assertEquals(20L, model.getSegmentID());
    }

    @Test
    void propertyTest() {
        Constraint model = new Constraint().property("tier");
        Assertions.assertEquals("tier", model.getProperty());
    }

    @Test
    void operatorTest() {
        Constraint model = new Constraint().operator(Constraint.OperatorEnum.EQ);
        Assertions.assertEquals(Constraint.OperatorEnum.EQ, model.getOperator());
    }

    @Test
    void valueTest() {
        Constraint model = new Constraint().value("premium");
        Assertions.assertEquals("premium", model.getValue());
    }

}

File: sdk/dart/test/constraint_test.dart
import 'package:test/test.dart';
import 'package:flagent_client/flagent_client.dart';

// tests for Constraint
void main() {
  final instance = Constraint((b) => b
    ..id = 1
    ..segmentID = 10
    ..property = 'region'
    ..operator_ = ConstraintOperator_Enum.valueOf('EQ')
    ..value = 'EU');

  group(Constraint, () {
    test('to test the property `id`', () async {
      expect(instance.id, equals(1));
    });

    test('to test the property `segmentID`', () async {
      expect(instance.segmentID, equals(10));
    });

    test('to test the property `property`', () async {
      expect(instance.property, equals('region'));
    });

    test('to test the property `operator_`', () async {
      expect(instance.operator_, equals(ConstraintOperator_Enum.valueOf('EQ')));
    });

    test('to test the property `value`', () async {
      expect(instance.value, equals('EU'));
    });
  });
}
